src/App.js

```diff
@@ -1,49 +1,96 @@
+/* global fetch */
+
 import React, { Component } from 'react'
-import {
-  FormGroup,
-  FormControl
-} from 'react-bootstrap'
 import './App.css'
-import PasteInlineForm from './pasteInlineForm.js'
+import PastePage from './PastePage'
+import {
+  BrowserRouter as Router,
+  Route
+} from 'react-router-dom'
 
-class App extends Component {
-  constructor (props) {
-    super(props)
+class TagsPage extends Component {
+  render () {
+    return (
+      <div>
+        <p>the tags page</p>
+      </div>
+    )
+  }
+}
+
+class DisplayPage extends Component {
+  constructor () {
+    super()
 
     this.state = {
-      codeText: '',
-      codeTitle: '',
-      codeAuthor: ''
+      codeText: ''
+    }
+  }
+
+  ifFbKeyOrNot (aStr) {
+    const patt = '^-[a-zA-Z0-9]+'
+
+    if (aStr.match(patt)) {
+      return aStr
+    } else {
+      return false
     }
+  }
+
+  componentWillMount () {
+    /* idk why eslint cant deal with assigning in an if */
+    let fbUrl = ''
+    const unclean = this.props.match.params.snipKey
+
+    const snipKey = this.ifFbKeyOrNot(unclean)
+
+    if (snipKey !== false) {
+      // 'https://console.firstproj-9f9e1.firebaseio.com/database/data/snip.json'
+      fbUrl = `https://firstproj-9f9e1.firebaseio.com/snip/${snipKey}.json`
+    } else {
+      this.setState({ codeText: 'bad url' })
+      return
+    }
+
+    // const getInit = {
+    //   method: 'GET'
+    // }
 
-    this.handleTextChange = this.handleTextChange.bind(this)
+    fetch(fbUrl)
+      .then(resp => resp.json())
+      .then((resp) => {
+        this.setState({ codeText: resp.codeText })
+      })
+      .catch((err) => {
+        console.log(err)
```

Comment:
`console.log()` messages in the code: this should not be passing lint without `// eslint-disable-line` ... what's happening here?